File: src/lessons/sort/basic/cocktail/AlgCocktailSort2Entity.c
//RemoteSort

/* BEGIN TEMPLATE */
void cocktailSort2() {
	/* BEGIN SOLUTION */
	int swapped;
	int begin=0;
	int end=getValueCount()-2;
	do {
		swapped = 0;
		int i;
		for (i=begin; i<=end; i++)
			if (!isSmaller(i,i+1)) {
				swap(i,i+1);
				swapped =1;
			}
		end--;
		for (i=end; i>=begin; i--)
			if (!isSmaller(i,i+1)) {
				swap(i,i+1);
				swapped =1;
			}
		begin++;
	} while (swapped && end-begin>1);
	/* END SOLUTION */
}
/* END TEMPLATE */


void run(){
	cocktailSort2();
}

File: resources/exercises/sort/basic/cocktail/AlgCocktailSort2Entity.py
# BEGIN SOLUTION 
begin = 0
end = getValueCount()-2
swapped = True
while swapped  and   end-begin>1:
	swapped = False
	for i in range(begin, (end+1) ):
		if not isSmaller(i,i+1):
			swap(i,i+1)
			swapped = True
	end -= 1
	for i in range(end,begin-1,-1):
		if not isSmaller(i,i+1):
			swap(i,i+1)
			swapped = True
	begin += 1
# END SOLUTION
